web-ng/src/app/components/layer-dialog/layer-dialog.component.ts

```diff
@@ -276,4 +291,35 @@ export class LayerDialogComponent implements OnDestroy {
       question?.questionInput?.nativeElement.focus();
     }
   }
+
+  private isFieldOptionsDirty(
+    formFieldEditor: FormFieldEditorComponent
+  ): boolean {
+    if (!formFieldEditor.optionEditors) {
+      return true;
+    }
+    for (const editor of formFieldEditor.optionEditors) {
+      if (editor.optionGroup.dirty) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  private hasUnsavedChanges(): boolean {
+    if (!this.formFieldEditors) {
+      return false;
+    }
+    for (const editor of this.formFieldEditors) {
+      if (editor.formGroup.dirty || !this.isFieldOptionsDirty(editor)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  private onClose(): void {
+    this.dialogRef.close();
+    return this.navigationService.selectProject(this.projectId!);
```

Comment:
Please add TODO to replace this with a new method in `NavigationService`, something like `closeLayerDialog()`. This method just has to remove the `#l=` part of the URL.